namer/consul/src/main/scala/io/buoyant/namer/consul/DcServices.scala

```diff
@@ -62,7 +62,14 @@ private[consul] object DcServices {
             // may consider retrying certain failures (with backoff).
             state() = Activity.Failed(e)
             stats.errors.incr()
-            Future.exception(e)
+            log.debug("consul connection failed: %s", name)
+
+            e match {
```

Comment:
I'm not 100% sure what's going on here, but when `ConnectionFailedException` happens (or connection refused or connection time out), `BaseApi` doesn't seem to call the backoff retry policy at all. I am sort of assuming the exception is happening in `client` and then the retry filter and api filter are skipped.

So this is a long way of saying I couldn't get it to work by adding something like
```
case (req, Throw(Failure(Some(err: ConnectionFailedException)))) => true
```
but I'm open to suggestions. I guess I could dig around in finagle to better understand how `Http.client` works...